ffi/cpp/include/writeable_utils.hpp

```diff
@@ -37,6 +37,18 @@ inline ICU4XWriteable WriteableFromString(std::string& string) {
   return w;
 };
 
+template<typename T> struct WriteableTrait {
+  // static inline ICU4XWriteable Construct(T& t);
+};
+
+
+template<> struct WriteableTrait<std::string> {
+  static inline ICU4XWriteable Construct(std::string& t) {
+    return WriteableFromString(t);
+  }
+};
```

Comment:
Question: How does someone outside ICU4X add an overload to this trait so that they can use your `FormatToWriteable` function?

Reply:
They write another `template<>` specialization like this one, except with their type instead of `std::string`

Reply:
What I'm getting at is, you suggest that they need to depend on something inside `icu4x::internal` ?

Reply:
Oh. I can put this in a different namespace.